container.ts

```diff
@@ -14,13 +14,27 @@ export interface Container<H extends Host> {
   executor: Executor;
 }
 
+export interface ExecutorDependencies {
+  config: Config;
+  editor: Editor;
+  host: Host;
+}
+
+export type ExecutorFactory = ({
+  config,
+  editor,
+  host,
+}: ExecutorDependencies) => Executor;
+
 export function container<H extends Host>(
   config: Config,
   host: H,
+  executorFactory: ExecutorFactory = ({ config, editor, host }) =>
```

Comment:
I'd love it if I could parametrize container by a subclass of Executor, the same way I do Host. But I can't both pass a default factory like this *and* introduce a parameter on ExecutorFactory.

One way to get around this could be to drop the factory and inject a class, but then I'm having to enforce types on the constructor - that I don't want to do.

Probably the right move is to do an unsafe cast in the relevant test helper(s).